### PR TITLE
bandwhich: init at 0.6.0

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/imsnif/what/releases/tag/0.6.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/76689)

### CODE REVIEW

**Comment by filalex77** [comment]
@GrahamcOfBorg eval

**Comment by filalex77** [comment]
Testing build on darwin
@GrahamcOfBorg build what

File: pkgs/tools/networking/what/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, fetchFromGitHub, rustPlatform }:
```
Comment (cust0dian) [inline pkgs/tools/networking/what/default.nix]:
```suggestion
{ darwin, lib, fetchFromGitHub, rustPlatform, stdenv }:
```

File: pkgs/tools/networking/what/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "imsnif";
+    repo = pname;
+    rev = version;
+    sha256 = "1q926w6c4hhf6gim6dn3jfcibgj5zbsgwnf5crmh8wv6a8fg6pxg";
+  };
+
+  cargoSha256 = "1rixpljqddwhryddzni5l6m4sjyn1krrj0ig0rzc701am7srhg3a";
+
```
Comment (cust0dian) [inline pkgs/tools/networking/what/default.nix]:
```suggestion

  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```

**Review by cust0dian-old — COMMENTED** [review]
Build on Darwin fails due to missing Security; I've added suggestions below on how to add a dependency on it (let me know if you'd rather see a PR to your fork).

After applying these the build seems to finish successfully (although `what` itself seems to be broken under macOS right now: https://github.com/imsnif/what/issues/31).

**Comment by teto** [comment]
seems like it got renamed to https://github.com/imsnif/bandwhich

**Comment by filalex77** [comment]
@cust0dian @teto Thanks, updated!

File: nixos/modules/programs/bandwhich.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [ bandwhich ];
+    security.wrappers.bandwhich = {
+      source = "${pkgs.bandwhich}/bin/bandwhich";
+      capabilities = "cap_net_raw,cap_net_admin+ep";
```
Comment (timokau) [inline nixos/modules/programs/bandwhich.nix]: What does the `+ep` mean here?

Comment (Mic92) [reply nixos/modules/programs/bandwhich.nix]: It adds both capabilities to the effective and permitted capability set: https://jlk.fjfi.cvut.cz/arch/manpages/man/capabilities.7#Thread_capability_sets

Which means that the current process can use those.

Comment (timokau) [reply nixos/modules/programs/bandwhich.nix]: Right, thank you

**Review by timokau — APPROVED** [review]
LGTM

**Pull request merged**